### PR TITLE
Feature 7

### PR DESCRIPTION
**Pull request by Bobreshovr**

### CODE REVIEW

**Comment by Bobreshovr** [comment]
https://github.com/TinkoffCreditSystems/Horarium/issues/7

File: README.md [hunk 1]
```diff
@@ -9,12 +9,21 @@ Horarium is fully based on an asynchronous work model, it allows you to run hund
 
 Horarium supports .NET Core/netstandard 2.0 and .NET Framework 4.6.2 and later.
 
+Support Database
```
Comment (mmikirtumov) [inline README.md]: Support Databases

Comment (Bobreshovr) [reply README.md]: Done

File: src/Horarium.Mongo/JobMongoModel.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
+using System;
+using Horarium.Repository;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+using MongoDB.Bson.Serialization.Options;
+
+namespace Horarium.Mongo
+{
+    [MongoEntity("scheduler.jobs")]
```
Comment (Bounz) [inline src/Horarium.Mongo/JobMongoModel.cs]: Maybe we should name this collection `horarium.jobs` (and collection for recurrent jobs too)?

File: src/Horarium.IntegrationTest/TestParallelsWorkTwoManagers.cs [hunk 3]
```diff
@@ -18,14 +16,14 @@ public class TestParallelsWorkTwoManagers : IntegrationTestBase
         public TestParallelsWorkTwoManagers()
         {
             var provider = new MongoClientProvider(ConnectionMongo);
-            var collection = provider.GetCollection<JobDb>();
+            var collection = provider.GetCollection<JobMongoModel>();
```
Comment (Bounz) [inline src/Horarium.IntegrationTest/TestParallelsWorkTwoManagers.cs]: Why integration test is stuck to Mongo DB provider?
Maybe we should rename the tests to reflect that these tests are for MongoDB?
Will be there same integration tests for other DB providers?

File: src/Horarium.Mongo/JobMongoModel.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                Status = Status,
+                JobType = JobType,
+                JobParamType = JobParamType,
+                JobParam = JobParam,
+                CountStarted = CountStarted,
+                StartedExecuting = StartedExecuting,
+                ExecutedMachine = ExecutedMachine,
+                StartAt = StartAt,
+                NextJob =
+                    NextJob?.ToJobDb(),
```
Comment (Bounz) [inline src/Horarium.Mongo/JobMongoModel.cs]: New line?

File: src/Horarium.Mongo/JobMongoModel.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                Cron = Cron,
+                Delay = Delay,
+                ObsoleteInterval = ObsoleteInterval
+            };
+        }
+
+        [BsonId] public string JobId { get; set; }
```
Comment (Bounz) [inline src/Horarium.Mongo/JobMongoModel.cs]: Maybe place attribute on a new line?

File: README.md [hunk 6]
```diff
@@ -9,12 +9,21 @@ Horarium is fully based on an asynchronous work model, it allows you to run hund
 
 Horarium supports .NET Core/netstandard 2.0 and .NET Framework 4.6.2 and later.
 
+Support Databases
+
+| Database   | Support                                                            |
+| ---------- | ------------------------------------------------------------------ |
+| MongoDB    | Yes                                                                |
+| In Memory  | No [#5](https://github.com/TinkoffCreditSystems/Horarium/issues/5) |
```
Comment (Bounz) [inline README.md]: Let's write `Not yet` instead of `No` 🙂 .

File: src/Horarium.Mongo/JobMongoModel.cs (hunk 2 above)
Comment (mmikirtumov) [reply src/Horarium.Mongo/JobMongoModel.cs]: what about horarium.jobs.scheduler?

File: src/Horarium.Mongo/Horarium.Mongo.csproj [hunk 7]
```diff
@@ -0,0 +1,22 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFrameworks>netstandard2.0</TargetFrameworks>
+        <Description>Horarium.Mongo</Description>
+        <Version>1.0.0$(VersionSuffix)</Version>
+        <Title>Horarium.Mongo</Title>
+        <Authors>bobreshovr</Authors>
+        <PackageDescription>Horarium.Mongo is the .Net library for support MongoDB in Horarium</PackageDescription>
```
Comment (mmikirtumov) [inline src/Horarium.Mongo/Horarium.Mongo.csproj]: Horarium.Mongo is the .Net library for support MongoDB in Horarium

May be?

> to support
or
> for supporting

File: src/Horarium/Interfaces/IHorarium.cs [hunk 8]
```diff
@@ -30,6 +31,6 @@ public interface IHorarium: IDisposable
         /// <typeparam name="TJob">Type of job, job will create from factory</typeparam>
         /// <returns></returns>
         IRecurrentJobBuilder CreateRecurrent<TJob>(string cron) where TJob : IJobRecurrent;
-        
+
     }
 }
```
Comment (mmikirtumov) [inline src/Horarium/Interfaces/IHorarium.cs]: no new line

File: src/Horarium.Mongo/JobMongoModel.cs (hunk 2 above)
Comment (Bobreshovr) [reply src/Horarium.Mongo/JobMongoModel.cs]: I think ```horarium.jobs``` is better

File: src/Horarium.IntegrationTest/TestParallelsWorkTwoManagers.cs (hunk 3 above)
Comment (Bobreshovr) [reply src/Horarium.IntegrationTest/TestParallelsWorkTwoManagers.cs]: I should exist this test for all providers, I will add TestCase

File: src/Horarium.Mongo/JobMongoModel.cs (hunk 4 above)
Comment (Bobreshovr) [reply src/Horarium.Mongo/JobMongoModel.cs]: Why not 😄

File: src/Horarium.Mongo/JobMongoModel.cs (hunk 2 above)
Comment (Bobreshovr) [reply src/Horarium.Mongo/JobMongoModel.cs]: done

File: src/Horarium.IntegrationTest/TestParallelsWorkTwoManagers.cs (hunk 3 above)
Comment (Bobreshovr) [reply src/Horarium.IntegrationTest/TestParallelsWorkTwoManagers.cs]: done

File: src/Horarium.Mongo/JobMongoModel.cs (hunk 4 above)
Comment (Bobreshovr) [reply src/Horarium.Mongo/JobMongoModel.cs]: done

File: src/Horarium.Mongo/JobMongoModel.cs (hunk 5 above)
Comment (Bobreshovr) [reply src/Horarium.Mongo/JobMongoModel.cs]: done

File: README.md (hunk 6 above)
Comment (Bobreshovr) [reply README.md]: done

File: src/Horarium.Mongo/Horarium.Mongo.csproj (hunk 7 above)
Comment (Bobreshovr) [reply src/Horarium.Mongo/Horarium.Mongo.csproj]: done

File: src/Horarium/Interfaces/IHorarium.cs (hunk 8 above)
Comment (Bobreshovr) [reply src/Horarium/Interfaces/IHorarium.cs]: done

**Pull request merged**